lib/MySQL_Session.cpp

```diff
@@ -4968,6 +4962,14 @@ int MySQL_Session::handler() {
 				gtid_hid = -1;
 				if (rc==0) {
 
+					if (active_transactions != 0) {  // run this only if currently we think there is a transaction
+						if ((myconn->mysql->server_status & SERVER_STATUS_IN_TRANS) == 0) { // there is no transaction on the backend connection
+							active_transactions = NumActiveTransactions(); // we check all the hostgroups/backends
+							if (active_transactions == 0)
+								transaction_started_at = 0; // reset it
+						}
+					}
+
```

Comment:
@renecannao what do you think about putting this code inside a function and calling it here and also when `rc==-1`. This way, ProxySQL also reset the timer when it has an error in the middle of a transaction.